webapp/src/components/call_error_modal.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {ComponentProps} from 'react';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import styled from 'styled-components';
+
+import GenericModal from 'src/components/generic_modal';
+import {clearClientError} from 'src/actions';
+import {getClientError} from 'src/selectors';
+
+import LaptopAlertSVG from 'src/components/icons/laptop_alert_svg';
+
+import {
+    rtcPeerErr,
+    rtcPeerCloseErr,
+    insecureContextErr,
+} from 'src/client';
+
+type Props = Partial<ComponentProps<typeof GenericModal>>;
+
+export const CallErrorModalID = 'call-error-modal';
+
+export const CallErrorModal = (props: Props) => {
+    const dispatch = useDispatch();
+    const clientErr = useSelector(getClientError);
+
+    if (!clientErr) {
+        return null;
+    }
+
+    const modalProps = {
+        ...props,
+    };
+
+    const onRejoinClick = (ev: React.MouseEvent) => {
+        ev.preventDefault();
+        window.postMessage({type: 'connectCall', channelID: clientErr.channelID}, window.origin);
+        dispatch(clearClientError());
+    };
+
+    const onConfirm = () => {
+        if (clientErr.err === insecureContextErr) {
+            window.open('https://docs.mattermost.com/configure/calls-deployment.html', '_blank');
+        }
+        return null;
+    };
+
+    const troubleShootingMsg = (
+        <React.Fragment>
+            {' Check the '}
```

Comment:
We'll probably want to localize these strings. 😅

Reply:
Yes, we want to localize everything before the end of next quarter but I am looking to have a dedicated epic for this work since we inevitably have to go through the existing code.